api/src/pcapi/core/users/email/update.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+from typing import Optional
+
+from flask import current_app as app
+from redis import Redis
+from sentry_sdk import capture_exception
+
+from pcapi import settings
+from pcapi.core.users import constants
+from pcapi.core.users import exceptions
+from pcapi.core.users import repository as users_repository
+from pcapi.core.users.models import User
+from pcapi.repository import user_queries
+
+from .send import send_user_emails_for_email_change
+
+
+def request_email_update(user: User, email: str, password: Optional[str]) -> None:
+    check_email_update_attempts(user, app.redis_client)
+    check_email_address_does_not_exist(email)
+    check_user_password(user, password)
+
+    expiration_date = save_email_update_activation_token_counter(user, app.redis_client)
+    send_user_emails_for_email_change(user, email, expiration_date)
+
+
+def check_email_update_attempts(user: User, redis: Redis) -> None:
+    update_email_attempts_key = f"update_email_attemps_user_{user.id}"
+    count = redis.incr(update_email_attempts_key)
+
+    if count == 1:
+        redis.expire(update_email_attempts_key, settings.EMAIL_UPDATE_ATTEMPTS_TTL)
+
+    if count > settings.MAX_EMAIL_UPDATE_ATTEMPTS:
+        raise exceptions.EmailUpdateLimitReached()
+
+
+def save_email_update_activation_token_counter(user: User, redis: Redis) -> datetime:
```

Comment:
```suggestion
def generate_token_expiration_date(user: User, redis: Redis) -> datetime:
```
?
Vu de l'extérieur de le counter ça a l'air d'être un détail d'implm
Tu peux éventuellement découper la partie "raise" dans un check_, mais l'utilité principale de cette fonction est bien d'avoir la date d'expiration non ?